core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -188,53 +197,82 @@
         final List<Closeable> resourcesToClose = new ArrayList<>(); // register everything we need to release in the case of an error
+        boolean success = false;
+        {
+            // use temp logger just to say we are starting. we can't use it later on because the node name might not be set
+            ESLogger logger = Loggers.getLogger(Node.class, NODE_NAME_SETTING.get(environment.settings()));
+            logger.info("initializing ...");
 
-        tmpSettings = TribeService.processSettings(tmpSettings);
-        ESLogger logger = Loggers.getLogger(Node.class, NODE_NAME_SETTING.get(tmpSettings));
-        final String displayVersion = Version.CURRENT + (Build.CURRENT.isSnapshot() ? "-SNAPSHOT" : "");
-        final JvmInfo jvmInfo = JvmInfo.jvmInfo();
-        logger.info(
-            "version[{}], pid[{}], build[{}/{}], OS[{}/{}/{}], JVM[{}/{}/{}/{}]",
-            displayVersion,
-            jvmInfo.pid(),
-            Build.CURRENT.shortHash(),
-            Build.CURRENT.date(),
-            Constants.OS_NAME,
-            Constants.OS_VERSION,
-            Constants.OS_ARCH,
-            Constants.JVM_VENDOR,
-            Constants.JVM_NAME,
-            Constants.JAVA_VERSION,
-            Constants.JVM_VERSION);
-
-        logger.info("initializing ...");
-
-        if (logger.isDebugEnabled()) {
-            logger.debug("using config [{}], data [{}], logs [{}], plugins [{}]",
-                    tmpEnv.configFile(), Arrays.toString(tmpEnv.dataFiles()), tmpEnv.logsFile(), tmpEnv.pluginsFile());
         }
-        // TODO: Remove this in Elasticsearch 6.0.0
-        if (JsonXContent.unquotedFieldNamesSet) {
-            DeprecationLogger dLogger = new DeprecationLogger(logger);
-            dLogger.deprecated("[{}] has been set, but will be removed in Elasticsearch 6.0.0",
+        try {
+            Settings tmpSettings = Settings.builder().put(environment.settings())
+                .put(Client.CLIENT_TYPE_SETTING_S.getKey(), CLIENT_TYPE).build();
+
+            tmpSettings = TribeService.processSettings(tmpSettings);
+
+            // create the node environment as soon as possible, to recovery the node id and enable logging
+            try {
+                nodeEnvironment = new NodeEnvironment(tmpSettings,environment);
+                resourcesToClose.add(nodeEnvironment);
+            } catch (IOException ex) {
+                throw new IllegalStateException("Failed to created node environment", ex);
+            }
+
+            final boolean hadPredefinedNodeName = NODE_NAME_SETTING.exists(tmpSettings);
+                tmpSettings = addNodeNameIfNeeded(tmpSettings, nodeEnvironment.nodeId());
+            ESLogger logger = Loggers.getLogger(Node.class, tmpSettings);
+            if (hadPredefinedNodeName == false) {
+                logger.info("node name set to [{}] by default. set the [{}] settings to change it",
```

Comment:
I think it's good to increase the visibility and encourage people to set their node names. I personally don't mind much to move it to debug.  Since @clintongormley asked for it, I'm pulling him in to get his opinion.